kiwi/config/functions.sh

```diff
@@ -998,11 +998,13 @@ function suseSetupProduct {
     # /etc/SuSE-brand or /etc/os-release or the latest .prod file
     # available in /etc/products.d
     # ----
+    set -e
     local prod=undef
     if [ -f /etc/SuSE-brand ];then
         prod=$(head /etc/SuSE-brand -n 1)
     elif [ -f /etc/os-release ];then
-        prod=$(grep "^NAME" /etc/os-release | cut -d '=' -f 2 | cut -d '"' -f 2)
+        prod=$(python3 -c 'import re
+with open("/etc/os-release") as f: name = re.search(r"^NAME.*$", f.read(-1), re.M)[0]; print(name.split("=")[1].replace("\"", ""))')
```

Comment:
can we be sure python to be around in any case ? I think this has potential to fail. The detection of the product is also one of the rare cases were many people rely on in scripts. A change here will impact many image builds. Thus should be done with care. From the change here I cannot tell for sure if it leads to the same result and to be honest what values python+re+replace over grep+cut ?

I'm just asking for the benefit of the change with respect to some potential for regressions

Thanks

Reply:
Well, kiwi itself is implemented in python, so python **must** be present for kiwi to run?

Reply:
you overlooked that scripts are called chrooted. The python requirement for kiwi is on the host. The requirement you added with the scripts now forces the image to have python installed. It's questionable if we want that

Reply:
Argh, ok, that's not something that I want to achieve. I was assuming that python would be available from kiwi anyway…